argo-rollouts-demo/README.md

```diff
@@ -0,0 +1,229 @@
+```
+
+3. Install the Pixie metrics server onto your cluster with:
+
+```
+kubectl apply -f px-metrics.yaml
+```
+
+## Setup the Demo Application
+
+1. [Install](https://argoproj.github.io/argo-rollouts/installation/#controller-installation) Argo Rollouts onto your cluster with:
+
+```
+kubectl create namespace argo-rollouts
+kubectl apply -n argo-rollouts -f https://github.com/argoproj/argo-rollouts/releases/latest/download/install.yaml
+```
+
+2. [Install](https://argoproj.github.io/argo-rollouts/installation/#kubectl-plugin-installation) the Argo Rollouts `kubectl` plugin. To use Homebrew, run:
+
+```
+brew install argoproj/tap/kubectl-argo-rollouts
+
+```
+
+2. Apply the manifests (including the application [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) and [AnalysisTemplate](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/pixie-analysis.yaml)):
+
+```
+kustomize build . | kubectl apply -f -
+```
+
+3. Get the `EXTERNAL-IP` address for the `canary-demo-preview` service:
+
+```
+kubectl get svc canary-demo-preview
+```
+
+4. Navigate to the External IP in your browser to access the demo application front-end. Leave this open.
+
+<br>
+<img src=".readme_assets/front-end.png"  width="550" align="right">
+<br>
+
+Each of these colorful squares is a request the browser is making to the backend.
+
+The backend responds with a color that indicates which version of the app it is. In this case we're getting blue boxes for the application image with the `blue` tag set in the [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) yaml.
+
+The bar chart at the bottom represents the percentage of requests that were handled by the different backends (stable, canary). We will see this in action in a minute.
+
+<br clear="all">
+
+## Successful Canary Rollout
+
+1. Watch the rollout live with the following command. Leave this tab open.
+
+```
+kubectl argo rollouts get rollout canary-demo --watch
+```
+
+2. In another tab, modify the [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) application image to trigger an upgrade:
+
+```
+kubectl argo rollouts set image canary-demo "*=argoproj/rollouts-demo:yellow"
+```
+
+3. You should now see blue (stable) running alongside yellow (canary):
+
+<br>
+<img src=".readme_assets/successful-rollout.png" width="550" align="right">
+<br>
+
+On our first step, we direct 50% of traffic to the `blue` release and 50% to `yellow` release.
+
+Argo Rollouts splits traffic between versions by creating a new replica set that uses the same service object and the service will still split the traffic evenly across pods (new and old). In other words, controlling the number of pods controls the traffic percentage.
+
+Our analysis is measuring the HTTP error rate for the canary pods every 30 seconds.
```

Comment:
Read through the whole README, and it's not clear where Pixie plays into this... I think we need to add a point here to make it very clear that the Pixie metrics server is what is supplying this HTTP error rate.